cirq-core/cirq/ops/linear_combinations_test.py

```diff
@@ -1662,3 +1662,253 @@ def test_expectation_from_density_matrix_two_qubit_states():
         np.testing.assert_allclose(
             psum3.expectation_from_density_matrix(state, qubit_map=q_map_2), 0
         )
+
+
+def test_projector_sum_expectations():
```

Comment:
Can we split this into two tests, one for state vector and one for density matrix. Also maybe using `cirq.testing.random_supoerposition` again to add a little omre complexity here ?

Reply:
I decided to split in 3, instead of 2: get the matrix, expectation from state vector, expectation from density matrix.

Also, I decided to add coefficients to both projectors.